### PR TITLE
[STORM-3580] Processs -c option to storm.py for Rebalance

### PR DESCRIPTION
**Pull request by bipinprasad**
storm.py passes utf encoded "-c" parameters as csv jvm property "storm.options". This system property is not being used in rebalance client command when passing information to Nimbus. Change Rebalance to use this system property.

### CODE REVIEW

File: storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java [hunk 1]
```diff
@@ -3356,6 +3356,12 @@ public void rebalance(String topoName, RebalanceOptions options)
             if (subject != null) {
                 options.set_principal(subject.getPrincipals().iterator().next().getName());
             }
+            String topoId = toTopoId(topoName);
```
Comment (Ethanlm) [inline storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: I think the change here should not be part of this PR

Comment (bipinprasad) [reply storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: Rebalance will not work without the change below. Should the change to Nimus.java be part of a different Jira then?

Comment (bipinprasad) [reply storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: Reverted Nimbus.java rebalance() changes.

File: storm-core/src/jvm/org/apache/storm/command/Rebalance.java [hunk 2]
```diff
@@ -57,6 +58,15 @@ public static void main(String[] args) throws Exception {
         }
 
         Map<String, Object> confOverrides = (Map<String, Object>) cl.get("t");
+        Map<String, Object> jvmOpts = Utils.readCommandLineOpts(); // values in -Dstorm.options (originally -c in storm.py)
```
Comment (Ethanlm) [inline storm-core/src/jvm/org/apache/storm/command/Rebalance.java]: This is mostly a question:  Do we want to include default `storm.yaml` into this too? 

for example: 
https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/command/ShellSubmission.java#L35 
this reads `storm.yaml` and commandLineOpts

File: storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java (hunk 1 above)
Comment (Ethanlm) [reply storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: Thanks

File: storm-core/src/jvm/org/apache/storm/command/Rebalance.java (hunk 2 above)
Comment (Ethanlm) [reply storm-core/src/jvm/org/apache/storm/command/Rebalance.java]: Offline discussion with Bipin :

> I don’t think we want that - that would be a reset+. Since rebalance should an incremental change to the currently running Topo.
